ICSharpCode.Decompiler/CSharp/Transforms/CustomPatterns.cs

```diff
@@ -70,9 +77,10 @@ public LdTokenPattern(string groupName)
 		
 		public override bool DoMatch(INode other, Match match)
 		{
-			InvocationExpression ie = other as InvocationExpression;
-			if (ie != null && ie.Annotation<LdTokenAnnotation>() != null && ie.Arguments.Count == 1) {
-				return childNode.DoMatch(ie.Arguments.Single(), match);
+			// new ErrorExpression("/*ldmembertoken ...*/")
+			if (other is ErrorExpression ee) {
```

Comment:
Instead of testing for `ErrorExpression`, you can check: `other is AstNode node && node.Annotation<LdMemberToken>() != null`